platform/source/mbed_retarget.cpp

```diff
@@ -768,13 +861,31 @@ extern "C" int PREFIX(_read)(FILEHANDLE fh, unsigned char *buffer, unsigned int
 
 extern "C" ssize_t read(int fildes, void *buf, size_t length)
 {
+    if (length == 0) {
+        return 0;
+    }
+
+#if MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY
+    if (fildes != STDIN_FILENO && fildes != STDERR_FILENO) {
+        return EBADF;
+    }
+
+    unsigned char *buffer = static_cast<unsigned char *>(buf);
+
+    for (size_t i = 0; i < length; i++) {
```

Comment:
This is wrong - implementing `read` on a TTY `getc` you have to read exactly 1 character, like `DirectSerial` does.

(And that's why you need the `length == 0` check on `read` only - to avoid the blocking `getc` when nothing is wanted. But given a >1-character buffer, you don't wait to fill the buffer, you return as soon as you have something).

Reply:
Thanks for explaining.

Reply:
Note that a device with a buffer could return more than 1 character in a read() call, so it's not entirely pointless. But in this case where we just have a blocking `getc` to work with, it's simplest to just return 1 character. We'll get called again if they want more.